FakeXrmEasy.Shared/XrmFakedContext.cs

```diff
@@ -321,29 +328,26 @@ protected IOrganizationService GetFakedOrganizationService(XrmFakedContext conte
         /// <param name="fakedService"></param>
         public static void FakeExecute(XrmFakedContext context, IOrganizationService fakedService)
         {
-            A.CallTo(() => fakedService.Execute(A<OrganizationRequest>._))
```

Comment:
@nicolabiancolini this change is what I am talking about in #374 when i said 
>  I don't understand why the change from local functions to funcs was needed

because the implementation is the same and I don't see any benefit from this change.
Best regards,
Betim.

Reply:
Ok.
Previously I had updated the nuget package `FakeItEasy` to version `4.9.2` and I run tests such as [When_a_request_is_called_with_an_empty_listid_parameter_exception_is_thrown](https://github.com/jordimontana82/fake-xrm-easy/blob/f7a5b5feb87c73fc09c4b8991727e5e637ddd320/FakeXrmEasy.Tests.Shared/FakeContextTests/AddListMembersListRequestTests/Tests.cs#L46) the returned exception is not `FaultException<OrganizationServiceFault>` but rather `FakeItEasy.UserCallbackException` as the exception generated within the `ReturnsLazily` method is captured and masked so that the original will be available as inner.

Instead, if you execute the function within the `Invokes` method, the returned exception is the one you want.
However, this method does not allow you to return values, which is necessary. That's why I moved the code outside creating some funcs.

This is what happens with the version `4.9.2` currently the version used is `3.2.0` and I have not checked if this happens anyway.

I have left the change because it solves a possible future problem and does not change the behavior.

Best regards,
Nicola.